File: spec/java/src/io/kaitai/struct/spec/TestEnumForUnknownId.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.EnumForUnknownId;
import org.testng.annotations.Test;

import static org.testng.Assert.assertNull;

public class TestEnumForUnknownId extends CommonSpec {
    @Test
    public void testEnumForUnknownId() throws Exception {
        EnumForUnknownId r = EnumForUnknownId.fromFile(SRC_DIR + "fixed_struct.bin");

        assertNull(r.one());
    }
}

File: spec/rust/tests/test_enum_for_unknown_id.rs
use std::fs;

extern crate kaitai;
use self::kaitai::*;
mod formats;
use formats::enum_for_unknown_id::*;


#[test]
fn test_enum_for_unknown_id() {
    let bytes = fs::read("../../src/fixed_struct.bin").unwrap();
    let reader = BytesReader::new(&bytes);
    let mut r = EnumForUnknownId::default();

    if let Err(err) = r.read(&reader, None, Some(KStructUnit::parent_stack())) {
        println!("expected err: {:?}, exception: UnknownVariant", err);
    } else {
        panic!("no expected exception: UnknownVariant");
    }

    //assert_eq!(r.one().clone().to_owned() as u64, 80);
}
